chore!: Normalize repository, dropping node <10.13 support

sttk opened the pull request:
This PR updates this project:

* Updates dependency versions
* Introduces GitHub Actions
* Adds, changes and deletes some project files

sttk committed:
chore!: Normalize repository, dropping node <10.13 support

phated commented on LICENSE:
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright (c) 2015, 2017 Blaine Bublitz <blaine.bublitz@gmail.com>, Eric Schoffstall <yo@contra.io> and other contributors
+Copyright (c) 2015, 2017, 2021 Blaine Bublitz <blaine.bublitz@gmail.com>, Eric Schoffstall <yo@contra.io> and other contributors
```
```suggestion
Copyright (c) 2015, 2017, 2021 Blaine Bublitz <blaine.bublitz@gmail.com> and Eric Schoffstall <yo@contra.io>
```

phated commented on README.md:
```diff
@@ -29,20 +29,19 @@ Takes a version and array of ranges, returns the greatest satisfied range. Range
-[gitter-url]: https://gitter.im/gulpjs/gulp
-[gitter-image]: https://badges.gitter.im/gulpjs/gulp.svg
+<!-- prettier-ignore-start -->
+[range-support]: https://github.com/phated/sver-compat#range-support
```
This needs to be updated to the official `sver` module. The documentation about "sver-compat" need to be updated to reference the "sver" project.

phated commented on package.json:
```diff
@@ -9,32 +9,37 @@
+    "test": "nyc mocha --async-only"
   },
   "dependencies": {
     "sver-compat": "^1.5.0"
```
This needs to use `sver` instead of my fork.

phated reviewed (changes requested):
Just needs to be updated to use the main `sver` package instead of my fork with node 0.10 support.

sttk committed:
Update LICENSE

Co-authored-by: Blaine Bublitz <blaine.bublitz@gmail.com>

sttk committed:
Update: Replace sver-compat to sver

sttk commented:
@phated I've modified what you pointed out.

I replaced `sver-compat` to `sver` but a test about prerelease version caused an error. Therefore I changed the 3rd parameter of `SemverRange.match` from `true` to `false`, then the test was passed. I don't know whether this modification is correct.

phated committed:
chore: Add test for newer prerelease versions

phated committed:
name it alpha

phated commented:
Thank you for identifying that change and telling me. I've thought a lot about this and tested some things locally and I feel that our tests were testing our desired behavior—that a prerelease version is not covered by the final release, since there could be breaking changes.

However, there were bugs in sver-compat that weren't handling this correctly which I found while writing some tests against it. For example, `1.0.1-alpha.1` should also not be satisfied by `^1.0.0` because there might be changes that require a patch release in other packages to be compatible. I've added that test on this branch.

phated reviewed (approved):
Thanks for all your work on this! 🙇
